src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -2872,26 +2880,8 @@ namespace InternalHeatGains {
                         }
 
                         // Environmental class
-                        if (UtilityRoutines::SameString(AlphaName(10), "None")) {
-                            thisZoneITEq.Class = ITEClassNone;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A1")) {
-                            thisZoneITEq.Class = ITEClassA1;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A2")) {
-                            thisZoneITEq.Class = ITEClassA2;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A3")) {
-                            thisZoneITEq.Class = ITEClassA3;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "A4")) {
-                            thisZoneITEq.Class = ITEClassA4;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "B")) {
-                            thisZoneITEq.Class = ITEClassB;
-                        } else if (UtilityRoutines::SameString(AlphaName(10), "C")) {
-                            thisZoneITEq.Class = ITEClassC;
-                        } else {
-                            ShowSevereError(state, std::string{RoutineName} + itEqModuleObject + ": " + AlphaName(1));
-                            ShowContinueError(state, "Invalid " + state.dataIPShortCut->cAlphaFieldNames(10) + '=' + AlphaName(10));
-                            ShowContinueError(state, "Valid entries are None, A1, A2, A3, A4, B or C.");
-                            ErrorsFound = true;
-                        }
+                        thisZoneITEq.Class = getEnumerationValue(ITEClassNamesUC, UtilityRoutines::MakeUPPERCase(AlphaName(10)));
```

Comment:
So `thisZoneITEq.Class` should be of type `ITEClass`, not `int`. There are a number of reasons for doing things this way, but the main one is that it's "safe" to cast an `enum class` to an `int` but not the other way around, so if you keep things as `enum class` you are always casting in a safe direction.

Reply:
Setting `thisZoneITEq.Class` to `ITEClass` is the first way I tried. Later I changed to the current way. One of the reasons is that doing the former would need to move the enum class and the namesUC definitions to "DataHeatBalance.hh", which seems a much wider scope for these variables. I wonder which way might be the better choice here when trying to weight in this consideration as well?

BTW, it seems that `getEnumerationValues()` will natively return an `int` type value. If setting `thisZoneITEq.Class` to `enum`, it seems that an `int` to `enum` cast is actually needed here (isn't this against the rule of conversion direction?): 
```
 thisZoneITEq.Class = static_cast<int>(getEnumerationValue(ITEClassNamesUC, UtilityRoutines::MakeUPPERCase(AlphaName(10))));
```
But in the current code there is only a `enum` to `int` cast is needed:
```
(thisZoneITEq.Class == static_cast<int>(ITEClass::Invalid));
```